File: 0228-summary-ranges/0228-summary-ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> ans;
        int i = 0, n = nums.size();
        
        while (i < n){
            int start = nums[i];
            while (i + 1 < n && nums[i] + 1 == nums[i + 1]) ++i;
            
            if (start != nums[i]) ans.push_back(to_string(start)+"->"+to_string(nums[i]));
            else ans.push_back(to_string(nums[i]));
            ++i;
        }
        
        return ans;
    }
};

File: 0228-summary-ranges/0228-summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        ans, n, i = [], len(nums), 0
        
        while i < n:
            start = nums[i]
            while i + 1 < n and nums[i] + 1 == nums[i + 1]:
                i += 1
            
            if start != nums[i]:
                ans.append(f'{start}->{nums[i]}')
            else:
                ans.append(f'{nums[i]}')
        
            i += 1
        
        return ans